keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -200,6 +227,34 @@ Spec.Containers[i].ResourcesAllocated are used to determine if there is enough
 room to admit the Pod. Kubelet does not set Pod's ResourcesAllocated after
 admitting a new Pod.
 
+<<[UNRESOLVED]>>
+The above is confusing.  Does it mean that ResourcesAllocated will be set by
+someone other than Kubelet?  Who?  Why?
+
+Above you said 'limit access to ResourcesAllocated field such that only Kubelet
+can update this field' which is the opposite of 'kubelet does not set Pod's
+ResourcesAllocated after admitting' here.  I am very confused.
+
+Naively, I would expect it to be:
+
+Create:
+* user creates a pod with `resources`
+* scheduler schedules based on new pod `resources` and existing pods'
+  `allocated`
+* kubelet accepts the pod based on new pod `resources` and existing pods'
+  `allocated`
+* kubelet sets `allocated` = `resources` immediately (acts as an ACK)
+* kubelet sets `actual` = `resources` once created
+
+Update:
+* user updates a pod's `resources`
+* kubelet considers the new `resources` and other pods' `allocated`
+* if viable, kubelet sets `allocated` = `resources` immediately (ACK) and sets
+  `actual` = `resources` once resized
+* if not viable, kubelet does not change `allocated` and instead drops an event
+  and maybe a new status field or condition
```

Comment:
This is how it works - actual = ContainerStatus.Resources.
We had a PodCondition proposed at one point in the KEP, but removed it for simplicity - discussion thread here: https://github.com/kubernetes/enhancements/pull/686#discussion_r297384981

Reply:
I disagree with the logic there - it's been a common trend that "more information is generally better".  If kubelet has considered a resize request and decided it can't do that right now, we REALLY should tell users and consumers.  Otherwise users will be forced to go to kubelet logs to figure out why the heck their change was accepted by "the system" (the API) but isn't happening.

We could argue you don't need a condition if it is just for humans (e.g. load-balancer creation is just events) but actually, we're now considering more status for LB creation, too.  Just as an analogy.

Reply:
I'll bring this up again in the next sig-node meeting. I had a [lot of pushback](https://github.com/kubernetes/enhancements/pull/686#discussion_r301296005) for this, mainly due to concerns of putting a point-in-time result as a pod condition, and if it was fully reconstructible, and added complexity.

I let it go because I felt if my concerns around not having it were to materialize, it can always be added at a later point. 

Are you available next Tuesday July 7th at 10-10:30 am for sig-node meeting?

Reply:
Unfortunately I have 2 other SIG meetings at the same time.  I do agree that we can always add this, but I'd put money on the notion that we WILL need it, and the arguments don't hold water for me.

All conditions are point-in-time - that's why they have a timestamp.

All conditions are reconstructible - they are `status` after all.

This is kind of what they are FOR.  Something about this pod needs to be noted.

Reply:
No worries. I'll bring it up anyways - having your voice would help a lot!